src/scales/scale.category.js

```diff
@@ -89,7 +91,7 @@ export default class CategoryScale extends Scale {
 		if (index < 0 || index > ticks.length - 1) {
 			return null;
 		}
-		return me.getPixelForValue(index * me._numLabels / ticks.length + me.min);
+		return me.getPixelForValue(ticks[index].value);
```

Comment:
The comment on this method says it supports duplicate values. I don't think this would work in that case because we wouldn't know whether it was the first or second one. The one case where I think duplicates might make sense was if you had something like https://i.stack.imgur.com/WcQd1.png

Reply:
The example images have duplicate labels. Values are unique, as its the index to the labels array.